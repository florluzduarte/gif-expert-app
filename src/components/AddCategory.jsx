import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = ({ target }) => {
    setInputValue(target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    const inputValueTrimmed = inputValue.trim();
    if (inputValueTrimmed.length <= 1) return;
    onNewCategory(inputValueTrimmed);
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={handleChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
