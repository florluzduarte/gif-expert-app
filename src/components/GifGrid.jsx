import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <div key={category}>
      <h2>{category}</h2>

      {isLoading && <h3>Cargando im√°genes...</h3>}

      <div className="card-grid">
        {gifs.map((images) => (
          <GifItem key={images.id} {...images} />
        ))}
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
