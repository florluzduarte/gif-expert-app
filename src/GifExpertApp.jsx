import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch", "Dragon Ball"]);

  const onAddCategory = (newCategory) => {
    const minCategory = newCategory.toLowerCase().replace(/\s/g, "");
    const minCategories = categories.map((category) =>
      category.toLowerCase().replace(/\s/g, "")
    );
    if (minCategories.includes(minCategory)) return;
    setCategories([...categories, newCategory]);
  };

  return (
    <>
      {/* Title */}
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory onNewCategory={onAddCategory} />
      {/* <AddCategory
        onNewCategory={(newCategory) => onAddCategory(newCategory)}
      /> */}
      {/* <AddCategory setCategories={setCategories} prevCategories={categories} /> */}
      {/* <AddCategory setCategories={setCategories} /> */}

      <ol>
        {categories.map((category) => {
          return <li key={category}>{category}</li>;
        })}
      </ol>
    </>
  );
};
