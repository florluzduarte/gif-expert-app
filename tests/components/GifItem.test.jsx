import { GifItem } from "../../src/components";
import { render, screen } from "@testing-library/react";

describe("<GifItem /> unit testing", () => {
  const title = "Sailor Moon";
  const url = "https://unicornio.dev/";

  test("Match with snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Check if alt and src properties from img tag are equal to title and url constants", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Check if title is render correctly", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
