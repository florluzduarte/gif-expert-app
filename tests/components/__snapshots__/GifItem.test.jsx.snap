// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> unit testing Match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Sailor Moon"
      src="https://unicornio.dev/"
    />
    <p>
      Sailor Moon
    </p>
  </div>
</div>
`;
